components/infobox/wikis/stormgate/infobox_league_custom.lua

```diff
@@ -0,0 +1,250 @@
+	return args.status == 'cancelled' or Logic.readBool(_args.cancelled or Variables.varDefault('cancelled tournament'))
+end
+
+---@param args table
+---@return boolean
+function CustomLeague._checkFinished(args)
+	if _args.cancelled then
+		return true
+	end
+	if args.status == 'finished' then
+		return true
+	end
+
+	local finished = Logic.readBoolOrNil(args.finished)
+	if finished ~= nil then
+		return finished
+	end
+
+	local queryDate = _league:_cleanDate(args.edate) or _league:_cleanDate(args.date) or FALLBACK_DATE
+	if os.date('%Y-%m-%d') >= queryDate then
+		local data = mw.ext.LiquipediaDB.lpdb('placement', {
+			conditions = '[[pagename::' .. string.gsub(mw.title.getCurrentTitle().text, ' ', '_') .. ']] '
+				.. 'AND [[opponentname::!TBD]] AND [[placement::1]]',
+			query = 'date',
+			order = 'date asc',
+			limit = 1
+		})
+		if data and data[1] then
+			return true
+		end
+	end
+
+	return false
+end
+
+---@return WidgetInjector
+function CustomLeague:createWidgetInjector()
+	return CustomInjector()
+end
+
+---@param id string
+---@param widgets Widget[]
+---@return Widget[]
+function CustomInjector:parse(id, widgets)
+	if id == 'gamesettings' then
+		table.insert(widgets, Cell{name = 'Game Version', content = {CustomLeague._getGameVersion(_args)}})
+	elseif id == 'customcontent' then
+		if _args.player_number and _args.player_number > 0 then
+			Array.appendWith(widgets,
+				Title{name = 'Player Breakdown'},
+				Cell{name = 'Number of Players', content = {_args.raceBreakDown.total}},
+				Breakdown{content = _args.raceBreakDown.display, classes = { 'infobox-center' }}
+			)
+		end
+
+		--teams section
+		if (tonumber(_args.team_number) or 0) > 0 then
+			Array.appendWith(widgets,
+				Title{name = 'Teams'},
+				Cell{name = 'Number of Teams', content = {_args.team_number}}
+			)
+		end
+
+		--maps
+		if String.isNotEmpty(_args.map1) then
+			Array.appendWith(widgets,
+				Title{name = 'Maps'},
+				Center{content = {CustomLeague._mapsDisplay(_args.maps)}}
+			)
+		end
+	end
+
+	return widgets
+end
+
+---@param maps {link: string, displayname: string}[]?
+---@return string?
+function CustomLeague._mapsDisplay(maps)
+	if not maps then return end
+
+	return table.concat(
+		Array.map(maps, function(mapData)
+			return tostring(CustomLeague:_createNoWrappingSpan(
+				PageLink.makeInternalLink({}, mapData.displayname, mapData.link)
+			))
+		end),
+		'&nbsp;• '
+	)
+end
+
+---@param args table
+---@return string
+function CustomLeague._getGameVersion(args)
+	local game = '[[' .. args.game .. ']]'
+
+	if not args.patch then
+		return game
+	end
+
+	local gameVersion = game .. '<br/>[[' .. args.patch .. ']]'
+	if not args.epatch or args.epatch == args.patch then
+		return gameVersion
+	end
+
+	return gameVersion .. ' &ndash; [[' .. args.epatch .. ']]'
+end
+
+---@param args table
+function CustomLeague:defineCustomPageVariables(args)
+	--override var to standardize its entries
+	Variables.varDefine('tournament_game', args.game)
```

Comment:
Is this needed?
Commons is like `Variables.varDefine('tournament_game', string.lower(args.game or ''))`

Reply:
i am not a fan of lowercasing it
we have several modules (incl via commons) that expect non lower cased iirc

Reply:
We have #3637 to figure that out, afaik there's only 1 game on the wiki, so not really important to have this set?